pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -52,23 +53,25 @@ type Webhook struct {
 	namespaceSelector *metav1.LabelSelector
 	objectSelector    *metav1.LabelSelector
 	containerRegistry string
+	datadogConfig     config.Component
 }
 
 // NewWebhook returns a new Webhook
-func NewWebhook() *Webhook {
-	nsSelector, objSelector := labelSelectors()
+func NewWebhook(datadogConfig config.Component) *Webhook {
+	nsSelector, objSelector := labelSelectors(datadogConfig)
 
 	containerRegistry := mutatecommon.ContainerRegistry("admission_controller.agent_sidecar.container_registry")
 
 	return &Webhook{
 		name:              webhookName,
-		isEnabled:         pkgconfigsetup.Datadog().GetBool("admission_controller.agent_sidecar.enabled"),
-		endpoint:          pkgconfigsetup.Datadog().GetString("admission_controller.agent_sidecar.endpoint"),
+		isEnabled:         datadogConfig.GetBool("admission_controller.agent_sidecar.enabled"),
+		endpoint:          datadogConfig.GetString("admission_controller.agent_sidecar.endpoint"),
 		resources:         []string{"pods"},
 		operations:        []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
 		namespaceSelector: nsSelector,
 		objectSelector:    objSelector,
 		containerRegistry: containerRegistry,
+		datadogConfig:     datadogConfig,
```

Comment:
It is great to now have access to the `config.Component` in the `newWebhook()` function. However instead of keep a "pointer" to the config in the `Webhook` struct to get config option values later in other functions, all the the values should be extract in this function and be store in this `Webhook` struct. 

I will give more details in next comments